keras_cv/layers/preprocessing/fmix.py

```diff
@@ -0,0 +1,221 @@
+    fmix = keras_cv.layers.preprocessing.mix_up.FMix(0.5)
+    augmented_images, updated_labels = fmix({'images': images, 'labels': labels})
+    # output == {'images': updated_images, 'labels': updated_labels}
+    ```
+    """
+
+    def __init__(
+        self, alpha=0.5, decay_power=3, edge_softening=0.0, seed=None, **kwargs
+    ):
+        super().__init__(seed=seed, **kwargs)
+        self.alpha = alpha
+        self.decay_power = decay_power
+        self.edge_softening = edge_softening
+        self.seed = seed
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta, shape):
+        sample_alpha = tf.random.gamma(shape, 1.0, beta=alpha)
+        sample_beta = tf.random.gamma(shape, 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    @staticmethod
+    def fftfreq(signal_size, sample_spacing=1):
+        """This function returns the sample frequencies of a discrete fourier transform.
+        The result array contains the frequency bin centers starting at 0 using the
+        sample spacing.
+        """
+
+        results = tf.concat(
+            [
+                tf.range((signal_size - 1) / 2 + 1, dtype=tf.int32),
+                tf.range(-(signal_size // 2), 0, dtype=tf.int32),
+            ],
+            0,
+        )
+
+        return results / (signal_size * sample_spacing)
+
+    def apply_fftfreq(self, h, w):
+        # Applying the fourier transform across 2 dimensions (height and width).
+
+        fx = FMix.fftfreq(w)[: w // 2 + 1 + w % 2]
+
+        fy = FMix.fftfreq(h)
+        fy = tf.expand_dims(fy, -1)
+
+        return tf.math.sqrt(fx * fx + fy * fy)
+
+    def get_spectrum(self, freqs, decay_power, channel, h, w):
+        # Function to apply a low pass filter by decaying its high frequency components.
+
+        scale = tf.ones(1) / tf.cast(
+            tf.math.maximum(freqs, tf.convert_to_tensor([1 / tf.reduce_max([w, h])]))
+            ** decay_power,
+            tf.float32,
+        )
+
+        param_size = tf.concat(
+            [tf.constant([channel]), tf.shape(freqs), tf.constant([2])], 0
+        )
+        param = self._random_generator.random_normal(param_size)
+
+        scale = tf.expand_dims(scale, -1)[None, :]
+
+        return scale * param
+
+    def sample_mask_from_transform(self, decay, shape, ch=1):
+        # Sampling low frequency map from fourier transform.
+
+        freqs = self.apply_fftfreq(shape[0], shape[1])
+        spectrum = self.get_spectrum(freqs, decay, ch, shape[0], shape[1])
+        spectrum = tf.complex(spectrum[:, 0], spectrum[:, 1])
+
+        mask = tf.math.real(tf.signal.irfft2d(spectrum, shape))
+        mask = mask[:1, : shape[0], : shape[1]]
+
+        mask = mask - tf.reduce_min(mask)
+        mask = mask / tf.reduce_max(mask)
+        return mask
+
+    def binarise_mask(self, mask, lam, in_shape, edge_softening=0.0):
+        # Create the final mask from the sampled values.
+
```

Comment:
Let's remove these newlines after comments.  No need for them, other than that the newlines are great!